### ISSUE TITLE
SCSS: Parsing error with class with 1 hyphen and interpolation

### ISSUE THREAD

**Issue by mayrop** [tonyganch/gonzales-pe#171]
There seems to be a parsing error when I try to create a class like this `.-whatever` using interpolation. 

Inside `file.scss`:

```
    $name: 'whatever';

    .-whatever { // works
        width: 0;
    }

    .#{$name} { // works
        width: 0;
    }

    ._#{$name} { // works
        width: 0;
    }

    .--#{$name} { // works
        width: 0;
    }

    .-#{$name} { //  Please check validity of the block starting from line #83  Fatal
        width: 0;
    }
```

**Comment by stowball** [comment]
I have this issue throughout my codebase, too. Here's a use-case:

``` scss
$colors: (
    core: (
        default: #2d343e,
        primary: #00cdcb,
        text: #767676,
        warning: #f65445
    )
);

.u-color {
    @each $color in map-get($colors, core) {
        &-#{nth($color, 1)} {
            color: c(core nth($color, 1)) !important;
        }
    }
}
```

**Comment by colin-marshall** [comment]
@stowball there's a fix on the way for this in the AST, see https://github.com/sasstools/sass-lint/issues/644.

**Comment by tonyganch** [comment]
I believe this should be fixed in dev already, will be released with upcoming v4.0.
Related PR: https://github.com/tonyganch/gonzales-pe/pull/215
